src/operations.jl

```diff
@@ -5,25 +5,9 @@ hash(c::AbstractRGB, hx::UInt) = hash(blue(c), hash(green(c), hash(red(c), hx)))
 hash(c::TransparentRGB, hx::UInt) = hash(alpha(c), hash(blue(c), hash(green(c), hash(red(c), hx))))
 
 # rand
-typealias RandTypes{T,C} Union{AbstractRGB{T},
-                               TransparentRGB{C,T},
-                               AbstractGray{T},
-                               TransparentGray{C,T}}
-typealias RandTypesFloat{C,T<:AbstractFloat} RandTypes{T,C}
-
-rand{G<:AbstractGray}(::Type{G}) = G(rand())
-rand{G<:TransparentGray}(::Type{G}) = G(rand(), rand())
-rand{C<:AbstractRGB}(::Type{C}) = C(rand(), rand(), rand())
-rand{C<:TransparentRGB}(::Type{C}) = C(rand(), rand(), rand(), rand())
-function rand{C<:RandTypesFloat}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(eltype(C), (sizeof(C)÷sizeof(eltype(C)), sz...)), sz)
-end
-function rand{C<:RandTypes{N0f8}}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(UInt8, (sizeof(C), sz...)), sz)
-end
-function rand{C<:RandTypes{N0f16}}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(UInt16, (sizeof(C)>>1, sz...)), sz)
-end
+rand{T<:Colorant}(::Type{T}) = reinterpret(T, rand(eltype(T), div(sizeof(T), sizeof(eltype(T)))))[1]
+rand{T<:Colorant}(::Type{T}, sz::Dims) =
+      reinterpret(T, rand(eltype(T), (div(sizeof(T), sizeof(eltype(T))), sz...)), sz)
```

Comment:
Here and above: `rand(RGB)` works currently but won't when this is merged. Maybe consider
```julia
rand{T<:Colorant}(::Type{T}) = _rand(ccolor(T, base_colorant_type(T){Float64}))
```
and then having `_rand` be the "real" implementation.